File: backend/entity/SaleContract.go
package entity

import "gorm.io/gorm"

type SaleContract struct {
	gorm.Model

	Status string

	CustomerID uint      `json:"CustomerID"`
	Customer   *Customer `gorm:"foreignKey:CustomerID" json:"Customer"`

	// Corrected to match the Car model's foreign key
	CarID uint `json:"CarID"`
	Car   *Car `gorm:"foreignKey:CarID" json:"Car"`

	EmployeeID uint      `json:"EmployeeID"`
	Employee   *Employee `gorm:"foreignKey:EmployeeID" json:"Employee"`

	//for Appointment
	InspectionAppointments []InspectionAppointment `gorm:"foreignKey:SaleContractID"`

	//ส่งไป payment
	Payment []Payment `gorm:"foreignKey:SaleContractID"`
}

File: frontend/src/interface/Salecontract.ts
import type { SaleList } from "./CarSell";
import type { Employee } from "./Employee";

export interface SaleContract {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  DeletedAt?: string | null;

  SaleListID: number;
  SaleList?: SaleList | null;

  EmployeeID: number;
  Employee?: Employee | null;

  CustomerID: number;
  Customer?: unknown | null;

  InspectionAppointments?: unknown | null;
  Payment?: unknown | null;
}
